From 7fec980f4aae28d304cdfd2297622c03a4cef2b6 Mon Sep 17 00:00:00 2001
Message-Id: <7fec980f4aae28d304cdfd2297622c03a4cef2b6.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: Shashank Sharma <shashank.sharma@intel.com>
Date: Wed, 28 Jan 2015 18:06:43 +0530
Subject: [PATCH 33/95] MUST_REBASE [VPG]: video/adf: Fix interface detect
 state

ADF framework's present code hardcodes detect_status variable to
one, while collecting data of an interface. This causes HWC to
assume that the interface is always connected.

This is not true for hot pluggable interfaces like HDMI, DP etc.

This patch does:
1. Changes the hardcoding to 1 to intf->hotplug_detect
   which was already there in the code, but commented out.
2. Set approprite value to this hotplug_detect function
   while setting the preferred mode.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed.

Issue: GMINL-5519
Change-Id: I350584a5f09fb84afdbdf5eb8fe157e569a9c125
Signed-off-by: Shashank Sharma <shashank.sharma@intel.com>
---
 drivers/video/adf/adf_fops.c                  |    2 +-
 drivers/video/adf/intel/intel_adf_interface.c |    9 +++++++--
 2 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/drivers/video/adf/adf_fops.c b/drivers/video/adf/adf_fops.c
index e54e6ad..665a638 100644
--- a/drivers/video/adf/adf_fops.c
+++ b/drivers/video/adf/adf_fops.c
@@ -600,7 +600,7 @@ static int adf_intf_get_data(struct adf_interface *intf,
 
 	mutex_lock(&dev->client_lock);
 	read_lock_irqsave(&intf->hotplug_modelist_lock, flags);
-	data.hotplug_detect = 1; /* intf->hotplug_detect; */
+	data.hotplug_detect = intf->hotplug_detect;
 	modelist_size = min(data.n_available_modes, intf->n_modes) *
 			sizeof(intf->modelist[0]);
 	memcpy(modelist, intf->modelist, modelist_size);
diff --git a/drivers/video/adf/intel/intel_adf_interface.c b/drivers/video/adf/intel/intel_adf_interface.c
index 7f0de80..71a5f17 100644
--- a/drivers/video/adf/intel/intel_adf_interface.c
+++ b/drivers/video/adf/intel/intel_adf_interface.c
@@ -273,8 +273,13 @@ static int set_preferred_mode(struct intel_adf_interface *intf)
 	if (err)
 		goto out_err0;
 
-	/* Only send HPD event in case of HDMI */
-	if (pipe->type == INTEL_PIPE_HDMI)
+	/*
+	 * Only send HPD event in case of hot-pluggable displays but set
+	 * the interface detect status for every interface, as this will be
+	 * used for real time status check from usp.
+	 */
+	intf->base.hotplug_detect = true;
+	if (pipe->type == INTEL_PIPE_HDMI || pipe->type == INTEL_PIPE_DP)
 		adf_hotplug_notify_connected(&intf->base, modelist, n_modes);
 
 	return 0;
-- 
1.7.9.5

